probe-rs/src/debug/unit_info.rs

```diff
@@ -1000,37 +986,52 @@ impl UnitInfo {
                 child_variable.type_name = VariableType::Pointer(type_name);
                 self.process_memory_location(
                     debug_info,
-                    node.entry(),
+                    node,
                     parent_variable,
                     child_variable,
                     memory,
                     frame_info,
+                    // NOTE: surprisingly, as opposed to `void*`, this can be a `const void*`.
                 )?;
 
                 // This needs to resolve the pointer before the regular recursion can continue.
-                match node.entry().attr(gimli::DW_AT_type) {
-                    Ok(Some(data_type_attribute)) => match data_type_attribute.value() {
-                        // NOTE: surprisingly, as opposed to `void*`, this can be a `const void*`.
-                        gimli::AttributeValue::UnitRef(unit_ref) => {
-                            child_variable.variable_node_type = VariableNodeType::ReferenceOffset(
-                                self.debug_info_offset()?,
-                                unit_ref,
-                            );
+                match node.attr_value(gimli::DW_AT_type) {
+                    Ok(Some(gimli::AttributeValue::UnitRef(unit_ref))) => {
+                        if !cache.has_children(child_variable) {
+                            let mut referenced_variable =
+                                cache.create_variable(child_variable.variable_key, Some(self))?;
+
+                            referenced_variable.name = match &child_variable.name {
```

Comment:
If you move the above comment back, could you add a TODO here to make this code lang-dependent? (Or you can do it, too, but it's out of scope, I just need a reminder to do it)